features/send_message.feature

```diff
@@ -0,0 +1,26 @@
+Feature: User can send a message
+
+As a user
+In order to send emails to my friends
+I should be able to compose an email
+
+Background: 
+    Given I visit the "landing" page
+    And The following users exist
+        | name     | email                  | password      | password_confirmation |
+        | Per Hegg | per_hegg@hotmail.com   | pereriksson   | pereriksson           |
+        | Laura    | laura@gmail.com        | laurareale    | laurareale            |
+    When I click on "Login"    
+    And I fill in "Email" with "per_hegg@hotmail.com"
+    And I fill in "Password" with "pereriksson"
+    And I click on "Log in"
+    Then I should see "Signed in successfully."
```

Comment:
You can refactor line 13-17 with one single step, please look into the course documentation for details.

Reply:
```
Background:
    Given following users exists
        | name     | email                  | password      | password_confirmation |
        | Per Hegg | per_hegg@hotmail.com   | pereriksson   | pereriksson           |
        | Laura    | laura@gmail.com        | laurareale    | laurareale            |
    And I am logged in as "Per Hegg"
    And I am on the "inbox" page 
```
```
Given("I am logged in as {string}") do |name|
    user = User.find_by(name: name)
    login_as user, scope: :user
end
```
```
When("I am on the {string} page") do |page_path|
    case page_path 
    when 'sign up'
        visit new_user_registration_path
    when 'inbox'
        visit mailbox_inbox_path
    else 
        return false
    end
end
```